lib/charms/prometheus_k8s/v0/prometheus_scrape.py

```diff
@@ -300,21 +300,49 @@ def _on_scrape_targets_changed(self, event):
 logger = logging.getLogger(__name__)
 
 
+JOB_KEYS = {
+    "job_name",
+    "metrics_path",
+    "static_configs",
+    "scrape_interval",
+    "scrape_timeout",
+    "proxy_url",
+    "relabel_configs",
+    "metrics_relabel_configs",
+    "sample_limit",
+    "label_limit",
+    "label_name_length_limit",
+    "label_value_lenght_limit",
+}
+DEFAULT_JOB = {
+    "metrics_path": "/metrics",
+    "static_configs": [{"targets": ["*:80"]}],
+}
+
+
 def _sanitize_scrape_configuration(job) -> dict:
     """Restrict permissible scrape configuration options.
 
+    If job is empty then a default job is returned. The
+    default job is
+
+    ```
+    {
+        "metrics_path": "/metrics",
+        "static_configs": [{"targets": ["*:80"]}],
+    }
+    ```
+
     Args:
-        job: a dict containing a single Prometheus job_name
+        job: a dict containing a single Prometheus job
             specification.
 
     Returns:
         a dictionary containing a sanitized job specification.
     """
-    return {
-        "job_name": job.get("job_name"),
-        "metrics_path": job.get("metrics_path", "/metrics"),
-        "static_configs": job.get("static_configs", [{"targets": ["*:80"]}]),
-    }
+    sanitized_job = DEFAULT_JOB.copy()
+    sanitized_job.update({key: value for key, value in job.items() if key in JOB_KEYS})
+    return sanitized_job
```

Comment:
add `logger.debug` for dropped (invalid) keys?

Reply:
Could do but I plan to document the allowed keys explicitly so there should be no surprises that keys are dropped.

Reply:
I am in process of explicitly documenting the allowed keys so there should be no surprises that keys are not showing up. There is already a unit test to validate that only allowed keys are put into scrape job configuration.

Reply:
unit tests here won't bubble this up when used by another charm.
a log entry would be quite valuable for debugging when a user relates their charm and nothing happens.

Reply:
I just pushed the updated docstring with list of allowed keys. I fail to see what is the point of logging rejected keys when it is explicitly documented that only a fixed list of keys are allowed.